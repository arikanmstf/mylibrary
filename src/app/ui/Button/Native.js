// @flow
import React from 'react';
import { Button as NativeButton, Text } from 'native-base';
import Link from 'ui/Link/Native';

import type { Node } from 'react';
import type { ButtonProps } from './types';

const ButtonLink = (props: ButtonProps): Node => {
  const { to, ...other } = props;

  return (
    <Link
      to={to}
    >
      <Text {...other} />
    </Link>
  );
};

const Button = (props: ButtonProps): Node => {
  const {
    onClick, text, to, ...other
  } = props;

  const foo = () => {};
  const onPress = onClick || foo;
  const Component = to ? ButtonLink : Text;

  return (
    <NativeButton
      full
      onPress={onPress}
      {...other}
    >
      <Component
        to={to}
        {...other}
      >
        {text}
      </Component>
    </NativeButton>
  );
};

export default Button;
