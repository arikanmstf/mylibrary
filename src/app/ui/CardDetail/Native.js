/**
 * Native Component Template By => create-module script
 * @version 1.0.1
 *
 */

// @flow
import React, { Component } from 'react';
import debounce from 'lodash.debounce';
import { withNavigation } from 'react-navigation';
import {
  Share,
  TouchableWithoutFeedback,
  TouchableOpacity,
  ScrollView,
  Linking,
} from 'react-native';
import { Image } from 'ui/native';
import {
  Card,
  CardItem,
  Text,
  Icon,
  Left,
  Body,
  Right,
  ActionSheet,
} from 'native-base';
import { production } from 'constants/routes/createUrl';
import logger from 'helpers/logger';

import { setCardType, defaultProps } from './helpers';
import type { CardDetailProps } from './types';

export class CardDetail extends Component<CardDetailProps> {
  static defaultProps = defaultProps;

  setCardType() {
    const { card, isDetailed } = this.props;
    setCardType.bind(this)(card, isDetailed);
  }

  goToDetail = () => {
    const { card, navigation } = this.props;

    if (card && this.getDetailUrl) {
      logger.log('goToDetail');
      const url = this.getDetailUrl();
      navigation.navigate(url, { id: card.id });
    }
  };

  shareCard = () => {
    const { card } = this.props;
    Share.share({
      url: production()(this.getDetailUrl(card.id)),
    });
  };

  goTo = (option) => {
    const { navigation } = this.props;
    if (option.to) {
      navigation.navigate(option.to, { id: option.toId });
    }
  };

  goToAddToList = () => {
    const { card, navigation } = this.props;

    logger.log('goToAddToList');
    const url = this.addToListUrl();
    navigation.navigate(url, { id: card.id });
  };

  goToDownload = async () => {
    const { card } = this.props;
    if (card && card.downloadUrl) {
      const supported = await Linking.canOpenURL(card.downloadUrl);

      if (supported) {
        Linking.openURL(card.downloadUrl);
      } else {
        logger.log(`Can't open the download url: ${card.downloadUrl}`);
      }
    }
  };

  handleRenderMoreClick = () => {
    const options = this.moreOptions.map((option) => (option.label));
    ActionSheet.show(
      {
        options,
        cancelButtonIndex: options.length - 1,
        destructiveButtonIndex: options.length - 1,
      },
      (buttonIndex) => {
        const option = this.moreOptions[buttonIndex];
        this.goTo(option);
      }
    );
  };

  toggleFavorite(id: number) {
    const { toggleFavorite } = this.props;
    if (toggleFavorite) {
      toggleFavorite(id);
    }
  }

  toggleRead(id: number) {
    const { toggleRead } = this.props;
    if (toggleRead) {
      toggleRead(id);
    }
  }

  render() {
    const { card, isDetailed } = this.props;

    if (!card) {
      return null;
    }
    this.setCardType();
    const goToDetailDebounce = debounce(this.goToDetail, 800, { leading: true, trailing: false });
    const goToAddToListDebounce = debounce(this.goToAddToList, 800, { leading: true, trailing: false });
    const goToDownloadDebounce = debounce(this.goToDownload, 800, { leading: true, trailing: false });

    logger.log('render: CardDetail');

    return (
      <ScrollView>
        <Card>
          <CardItem>
            <Body>
              <Text>{card.title}</Text>
              <Text note>{card.subHeader}</Text>
            </Body>
            <Right
              style={{
                flex: 1,
                alignItems: 'center',
                flexDirection: 'row',
                justifyContent: 'flex-end',
              }}
            >
              { card && this.moreOptions.length > 1 ? (
                <TouchableOpacity
                  onPress={this.handleRenderMoreClick}
                  style={{ width: 40, flexShrink: 1, alignItems: 'flex-end' }}
                >
                  <Icon name="more" active />
                </TouchableOpacity>
              ) : null }
            </Right>
          </CardItem>
          { this.imageUri ? (
            <TouchableWithoutFeedback
              onPress={goToDetailDebounce}
            >
              <CardItem
                cardBody
              >
                <Image
                  source={{ uri: this.imageUri }}
                  style={{ height: 200, flex: 1 }}
                />
              </CardItem>
            </TouchableWithoutFeedback>
          ) : null }
          { isDetailed ? (
            <CardItem>
              <Body>
                <Text>{card.description}</Text>
              </Body>
            </CardItem>
          ) : null }
          <CardItem>
            <Left>
              { card.isFavorite !== undefined ? (
                <TouchableOpacity
                  style={{ width: 40 }}
                  onPress={() => { this.toggleFavorite(card.id); }}
                >
                  <Icon name="star" active={card.isFavorite} style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
              { card.isRead !== undefined ? (
                <TouchableOpacity
                  style={{ width: 40, marginTop: 3 }}
                  onPress={() => { this.toggleRead(card.id); }}
                >
                  <Icon name="book" active={card.isRead} style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
              { this.addToListUrl ? (
                <TouchableOpacity
                  style={{ width: 40, marginTop: 3 }}
                  onPress={goToAddToListDebounce}
                >
                  <Icon name="add" style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
            </Left>
            <Right
              style={{
                flex: 1,
                alignItems: 'center',
                flexDirection: 'row',
                justifyContent: 'flex-end',
              }}
            >
              { card.downloadUrl ? (
                <TouchableOpacity
                  style={{ width: 40 }}
                  onPress={goToDownloadDebounce}
                >
                  <Icon name="download" color="#000" style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
              { this.getDetailUrl ? (
                <TouchableOpacity
                  style={{ width: 40, flexShrink: 1, alignItems: 'flex-end' }}
                  onPress={this.shareCard}
                >
                  <Icon name="share" color="#000" style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
            </Right>
          </CardItem>
        </Card>
      </ScrollView>
    );
  }
}

export default withNavigation(CardDetail);
