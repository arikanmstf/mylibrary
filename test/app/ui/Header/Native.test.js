/**
 * Native Component Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import { Header, renderList, SideBar } from 'ui/Header/Native';
import routes from 'ui/Header/sideNavigationItems';

describe('test/app/ui/Header/Native.test.js', () => {
  const props = {};
  const wrapper = shallow(<Header {...props} />);

  describe('Render', () => {
    it('props: {}', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('props: { back: true }', () => {
      wrapper.setProps({ back: true });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('handleMenuButtonPress', () => {
    const showDrawer = jest.fn();
    const newProps = {
      showDrawer,
    };

    wrapper.setProps(newProps);
    wrapper.instance().handleMenuButtonPress();
    expect(showDrawer).toBeCalled();
  });

  it('renderList', () => {
    const route = routes[0];
    expect(renderList(route)).toMatchSnapshot();
  });

  it('SideBar', () => {
    const wrapperSidebar = shallow(<SideBar />);
    expect(wrapperSidebar).toMatchSnapshot();
  });

  it('renderBackButton', () => {
    expect(wrapper.instance().renderBackButton()).toMatchSnapshot();
  });

  it('renderCenter', () => {
    expect(wrapper.instance().renderCenter()).toMatchSnapshot();
  });
});
