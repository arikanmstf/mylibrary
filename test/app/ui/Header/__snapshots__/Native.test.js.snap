// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/ui/Header/Native.test.js Render props: { back: true } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    back={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Left)>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "width": 70,
              }
            }
          >
            <Styled(Icon)
              name="arrow-back"
            />
          </TouchableOpacity>
        </Styled(Left)>,
        <Styled(Body)>
          <Styled(Item)
            style={
              Object {
                "flexGrow": 1,
                "height": 25,
                "marginLeft": 20,
              }
            }
          >
            <Styled(Icon)
              name="ios-search"
            />
            <TextField
              label="Search"
              name="search"
              onChange={[Function]}
              type="search"
            />
          </Styled(Item)>
        </Styled(Body)>,
        <Styled(Right)>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "width": 70,
              }
            }
          >
            <Styled(Icon)
              name="menu"
            />
          </TouchableOpacity>
        </Styled(Right)>,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "width": 70,
              }
            }
          >
            <Styled(Icon)
              name="arrow-back"
            />
          </TouchableOpacity>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Styled(Icon)
              name="arrow-back"
            />,
            "onPress": [Function],
            "style": Object {
              "width": 70,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "arrow-back",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Item)
            style={
              Object {
                "flexGrow": 1,
                "height": 25,
                "marginLeft": 20,
              }
            }
          >
            <Styled(Icon)
              name="ios-search"
            />
            <TextField
              label="Search"
              name="search"
              onChange={[Function]}
              type="search"
            />
          </Styled(Item)>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Icon)
                name="ios-search"
              />,
              <TextField
                label="Search"
                name="search"
                onChange={[Function]}
                type="search"
              />,
            ],
            "style": Object {
              "flexGrow": 1,
              "height": 25,
              "marginLeft": 20,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "ios-search",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "Search",
                "name": "search",
                "onChange": [Function],
                "type": "search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "width": 70,
              }
            }
          >
            <Styled(Icon)
              name="menu"
            />
          </TouchableOpacity>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Styled(Icon)
              name="menu"
            />,
            "onPress": [Function],
            "style": Object {
              "alignItems": "flex-end",
              "flex": 1,
              "width": 70,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "menu",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Left)>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "width": 70,
                }
              }
            >
              <Styled(Icon)
                name="arrow-back"
              />
            </TouchableOpacity>
          </Styled(Left)>,
          <Styled(Body)>
            <Styled(Item)
              style={
                Object {
                  "flexGrow": 1,
                  "height": 25,
                  "marginLeft": 20,
                }
              }
            >
              <Styled(Icon)
                name="ios-search"
              />
              <TextField
                label="Search"
                name="search"
                onChange={[Function]}
                type="search"
              />
            </Styled(Item)>
          </Styled(Body)>,
          <Styled(Right)>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "width": 70,
                }
              }
            >
              <Styled(Icon)
                name="menu"
              />
            </TouchableOpacity>
          </Styled(Right)>,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "width": 70,
                }
              }
            >
              <Styled(Icon)
                name="arrow-back"
              />
            </TouchableOpacity>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Styled(Icon)
                name="arrow-back"
              />,
              "onPress": [Function],
              "style": Object {
                "width": 70,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "arrow-back",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Item)
              style={
                Object {
                  "flexGrow": 1,
                  "height": 25,
                  "marginLeft": 20,
                }
              }
            >
              <Styled(Icon)
                name="ios-search"
              />
              <TextField
                label="Search"
                name="search"
                onChange={[Function]}
                type="search"
              />
            </Styled(Item)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(Icon)
                  name="ios-search"
                />,
                <TextField
                  label="Search"
                  name="search"
                  onChange={[Function]}
                  type="search"
                />,
              ],
              "style": Object {
                "flexGrow": 1,
                "height": 25,
                "marginLeft": 20,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "ios-search",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Search",
                  "name": "search",
                  "onChange": [Function],
                  "type": "search",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "width": 70,
                }
              }
            >
              <Styled(Icon)
                name="menu"
              />
            </TouchableOpacity>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Styled(Icon)
                name="menu"
              />,
              "onPress": [Function],
              "style": Object {
                "alignItems": "flex-end",
                "flex": 1,
                "width": 70,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "menu",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`test/app/ui/Header/Native.test.js Render props: {} 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Left)>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "width": 70,
              }
            }
          >
            <Styled(Icon)
              name="arrow-back"
            />
          </TouchableOpacity>
        </Styled(Left)>,
        <Styled(Body)>
          <Styled(Item)
            style={
              Object {
                "flexGrow": 1,
                "height": 25,
                "marginLeft": 20,
              }
            }
          >
            <Styled(Icon)
              name="ios-search"
            />
            <TextField
              label="Search"
              name="search"
              onChange={[Function]}
              type="search"
            />
          </Styled(Item)>
        </Styled(Body)>,
        <Styled(Right)>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "width": 70,
              }
            }
          >
            <Styled(Icon)
              name="menu"
            />
          </TouchableOpacity>
        </Styled(Right)>,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "width": 70,
              }
            }
          >
            <Styled(Icon)
              name="arrow-back"
            />
          </TouchableOpacity>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Styled(Icon)
              name="arrow-back"
            />,
            "onPress": [Function],
            "style": Object {
              "width": 70,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "arrow-back",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Item)
            style={
              Object {
                "flexGrow": 1,
                "height": 25,
                "marginLeft": 20,
              }
            }
          >
            <Styled(Icon)
              name="ios-search"
            />
            <TextField
              label="Search"
              name="search"
              onChange={[Function]}
              type="search"
            />
          </Styled(Item)>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Icon)
                name="ios-search"
              />,
              <TextField
                label="Search"
                name="search"
                onChange={[Function]}
                type="search"
              />,
            ],
            "style": Object {
              "flexGrow": 1,
              "height": 25,
              "marginLeft": 20,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "ios-search",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "Search",
                "name": "search",
                "onChange": [Function],
                "type": "search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "width": 70,
              }
            }
          >
            <Styled(Icon)
              name="menu"
            />
          </TouchableOpacity>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Styled(Icon)
              name="menu"
            />,
            "onPress": [Function],
            "style": Object {
              "alignItems": "flex-end",
              "flex": 1,
              "width": 70,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "menu",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Left)>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "width": 70,
                }
              }
            >
              <Styled(Icon)
                name="arrow-back"
              />
            </TouchableOpacity>
          </Styled(Left)>,
          <Styled(Body)>
            <Styled(Item)
              style={
                Object {
                  "flexGrow": 1,
                  "height": 25,
                  "marginLeft": 20,
                }
              }
            >
              <Styled(Icon)
                name="ios-search"
              />
              <TextField
                label="Search"
                name="search"
                onChange={[Function]}
                type="search"
              />
            </Styled(Item)>
          </Styled(Body)>,
          <Styled(Right)>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "width": 70,
                }
              }
            >
              <Styled(Icon)
                name="menu"
              />
            </TouchableOpacity>
          </Styled(Right)>,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "width": 70,
                }
              }
            >
              <Styled(Icon)
                name="arrow-back"
              />
            </TouchableOpacity>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Styled(Icon)
                name="arrow-back"
              />,
              "onPress": [Function],
              "style": Object {
                "width": 70,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "arrow-back",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Item)
              style={
                Object {
                  "flexGrow": 1,
                  "height": 25,
                  "marginLeft": 20,
                }
              }
            >
              <Styled(Icon)
                name="ios-search"
              />
              <TextField
                label="Search"
                name="search"
                onChange={[Function]}
                type="search"
              />
            </Styled(Item)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(Icon)
                  name="ios-search"
                />,
                <TextField
                  label="Search"
                  name="search"
                  onChange={[Function]}
                  type="search"
                />,
              ],
              "style": Object {
                "flexGrow": 1,
                "height": 25,
                "marginLeft": 20,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "ios-search",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Search",
                  "name": "search",
                  "onChange": [Function],
                  "type": "search",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "width": 70,
                }
              }
            >
              <Styled(Icon)
                name="menu"
              />
            </TouchableOpacity>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Styled(Icon)
                name="menu"
              />,
              "onPress": [Function],
              "style": Object {
                "alignItems": "flex-end",
                "flex": 1,
                "width": 70,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "menu",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`test/app/ui/Header/Native.test.js SideBar 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(List)
        dataArray={
          Array [
            Object {
              "icon": "settings",
              "label": "HEADER_MENU_PROFILE",
              "to": "/profile",
            },
            Object {
              "icon": "star",
              "label": "HEADER_MENU_FAVORITES",
              "to": "/favorites",
            },
            Object {
              "icon": "book",
              "label": "HEADER_MENU_BOOKS_I_READ",
              "to": "/books-i-read",
            },
            Object {
              "icon": "log-out",
              "label": "HEADER_MENU_LOGOUT",
              "to": "/logout",
            },
          ]
        }
        renderRow={[Function]}
      />,
      "style": Object {
        "backgroundColor": "#ffffff",
        "paddingTop": 20,
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dataArray": Array [
          Object {
            "icon": "settings",
            "label": "HEADER_MENU_PROFILE",
            "to": "/profile",
          },
          Object {
            "icon": "star",
            "label": "HEADER_MENU_FAVORITES",
            "to": "/favorites",
          },
          Object {
            "icon": "book",
            "label": "HEADER_MENU_BOOKS_I_READ",
            "to": "/books-i-read",
          },
          Object {
            "icon": "log-out",
            "label": "HEADER_MENU_LOGOUT",
            "to": "/logout",
          },
        ],
        "renderRow": [Function],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(List)
          dataArray={
            Array [
              Object {
                "icon": "settings",
                "label": "HEADER_MENU_PROFILE",
                "to": "/profile",
              },
              Object {
                "icon": "star",
                "label": "HEADER_MENU_FAVORITES",
                "to": "/favorites",
              },
              Object {
                "icon": "book",
                "label": "HEADER_MENU_BOOKS_I_READ",
                "to": "/books-i-read",
              },
              Object {
                "icon": "log-out",
                "label": "HEADER_MENU_LOGOUT",
                "to": "/logout",
              },
            ]
          }
          renderRow={[Function]}
        />,
        "style": Object {
          "backgroundColor": "#ffffff",
          "paddingTop": 20,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dataArray": Array [
            Object {
              "icon": "settings",
              "label": "HEADER_MENU_PROFILE",
              "to": "/profile",
            },
            Object {
              "icon": "star",
              "label": "HEADER_MENU_FAVORITES",
              "to": "/favorites",
            },
            Object {
              "icon": "book",
              "label": "HEADER_MENU_BOOKS_I_READ",
              "to": "/books-i-read",
            },
            Object {
              "icon": "log-out",
              "label": "HEADER_MENU_LOGOUT",
              "to": "/logout",
            },
          ],
          "renderRow": [Function],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`test/app/ui/Header/Native.test.js renderBackButton 1`] = `
<Styled(Left)>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "width": 70,
      }
    }
  >
    <Styled(Icon)
      name="arrow-back"
    />
  </TouchableOpacity>
</Styled(Left)>
`;

exports[`test/app/ui/Header/Native.test.js renderCenter 1`] = `
<Styled(Body)>
  <Styled(Item)
    style={
      Object {
        "flexGrow": 1,
        "height": 25,
        "marginLeft": 20,
      }
    }
  >
    <Styled(Icon)
      name="ios-search"
    />
    <TextField
      label="Search"
      name="search"
      onChange={[Function]}
      type="search"
    />
  </Styled(Item)>
</Styled(Body)>
`;

exports[`test/app/ui/Header/Native.test.js renderList 1`] = `
<Styled(ListItem)
  button={true}
>
  <Styled(Icon)
    name="settings"
    style={
      Object {
        "marginRight": 20,
      }
    }
  />
  <Link
    component={[Function]}
    replace={false}
    to="/profile"
  >
    <Styled(Text)>
      Profile
    </Styled(Text)>
  </Link>
</Styled(ListItem)>
`;
